### PR TITLE
Category as first argument for `GeneralizedMorphismCategoryByCospans`

### PR DESCRIPTION
**Pull request by TKuh**
Part of #973.

### CODE REVIEW

**Review by zickgraf — CHANGES_REQUESTED** [review]
Thanks! One small change request below, and a general comment in case you are not aware: setting `category_as_first_argument` is completely independent of calling categorical operations with the category as first argument, i.e. it would be enough to simply add a new argument to the functions added via `Add...` without modifying their body. Of course calling categorical operations with the category as first argument is also nice especially if we want to compile the code in the future. But I expect that to take much more time

**Comment by TKuh** [comment]
> setting `category_as_first_argument` is completely independent of calling categorical operations with the category as first argument, i.e. it would be enough to simply add a new argument to the functions added via `Add...` without modifying their body. 

Thanks for the clarification. I suspected it but wasn't 100% sure. :)

> Of course calling categorical operations with the category as first argument is also nice especially if we want to compile the code in the future. But I expect that to take much more time

My reasoning was, that in this case it's fine because of issue #941. But I will not spend my time on these things in the future then, unless something has to be compilable eventually.

File: GeneralizedMorphismsForCAP/gap/GeneralizedMorphismCategoryByCospans.gi [hunk 1]
```diff
@@ -19,35 +19,51 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_GENERALIZED_MORPHISM_CATEGORY_BY_CO
     underlying_honest_category := UnderlyingHonestCategory( category );
     
     ##
-    AddIsEqualForCacheForObjects( category, IsIdenticalObj );
```
[the remark this answers is not in the document]
Comment (TKuh) [reply GeneralizedMorphismsForCAP/gap/GeneralizedMorphismCategoryByCospans.gi:22]: Done, thanks for the explanation.

**Review by zickgraf — APPROVED** [review]
Thanks!

**Comment by zickgraf** [comment]
> > Of course calling categorical operations with the category as first argument is also nice especially if we want to compile the code in the future. But I expect that to take much more time
> 
> My reasoning was, that in this case it's fine because of issue #941. But I will not spend my time on these things in the future then, unless something has to be compilable eventually.

I agree!

**Pull request merged**